### PR TITLE
logging: Print iTerm link to the log's source file:line in VS Code

### PR DESCRIPTION
**Pull request by chrismwendt**
This prints a little iTerm link to the log statement's source file:line in VS Code.

Enable it by setting `SRC_LOG_SOURCE_LINK=true`.

More terminal+editor combinations could be supported in the future.

https://user-images.githubusercontent.com/1387653/151882112-7e3b06e7-d0ba-42a8-9cd1-960f8640be82.mp4

### CODE REVIEW

**Review by eseliger — APPROVED** [review]
very cool!

File: internal/logging/main.go [hunk 1]
```diff
@@ -93,6 +95,16 @@ func LogEntryLevelString(l log15.Lvl) string {
...
 }
 
+// From iTerm section "Anchor (OSC 8)"
+// https://iterm2.com/documentation-escape-codes.html
+func itermLink(url, text string) string {
+	return fmt.Sprintf("\x1B]8;;%s\x07%s\x1B]8;;\x07", url, text)
+}
+
+func vscodeUrl(file string) string {
+	return fmt.Sprintf("vscode://file/%s", file)
```
Comment (felixfbecker) [inline internal/logging/main.go:105]: If this used a `file://` URL instead, would that just open the default editor for the file type (which can be VS Code, but also e.g. Goland)?

Comment (chrismwendt) [reply internal/logging/main.go:105]: I think `file://` can only link to a file, not file:line.

The only way to open a file at a line number with GoLand I've seen so far is by running `goland --line 42 ...`, which  https://www.jetbrains.com/help/go/opening-files-from-command-line.html

It looks like iTerm only supports opening files/URLs with Cmd+Click, not running arbitrary commands:

CleanShot 2022-01-31 at 17 08 48@2x

Registering a custom handler might work https://superuser.com/questions/548119/how-do-i-configure-custom-url-handlers-on-os-x

**Review by mrnugget — APPROVED** [review]
How about adding adding `SRC_LOG_SOURCE_LINK=false` to `sg.config.yaml` with a comment so that it's easier to discover?

**Pull request merged**